mat-process-api/V1/Gateways/ImagePersistingGateway.cs

```diff
@@ -46,18 +46,13 @@ public string RetrieveImage(string imageKey)
             {
                 var result = s3Client.retrieveImage(assumeRoleHelper.GetTemporaryCredentials(), imageKey,
                     Environment.GetEnvironmentVariable("bucket-name"));
-
-                if(result.HttpStatusCode != HttpStatusCode.OK)
-                {
-                    throw new ImageNotFound();
-                }
                 return ImageRetrievalFactory.EncodeStreamToBase64(result);
             }
             catch(AggregateException ex)
             {
-                if(ex.Message == "The specified key does not exist.")
+                if(ex.InnerException != null && ex.InnerException.Message == "The specified key does not exist.")
```

Comment:
I understand what you tried doing here, I'm wondering though whether this would work?

![image](https://user-images.githubusercontent.com/43747286/72911821-996df200-3d32-11ea-8923-3bf92f05e7b3.png)
[Try-Catch Documentation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/try-catch)
I'm wondering whether you can check the type of inner exception in the `when` statement:
Like:
`catch (AggregateException ex) when (typeof(ex.InnerException) == typeof(ImageNotFound))` or something among these lines.

Reply:
That way we won't need the hacky looking re-throw of exception that doesn't match the criteria.

Reply:
Just tried it, all the tests regarding this implementation pass:

![image](https://user-images.githubusercontent.com/43747286/72913210-b6a3c000-3d34-11ea-88eb-8e18da73398b.png)